src/core/allow-bind.c

```diff
@@ -0,0 +1,164 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#if HAVE_LIBBPF
+#include <bpf/bpf.h>
+#endif
+
+#include "allow-bind.h"
+#include "fd-util.h"
+#include "unit.h"
+
+#if BPF_FRAMEWORK
+/* libbpf, clang and llc compile time dependencies are satisfied */
+#include "bpf/allow_bind/allow-bind-skel.h"
+
+static struct allow_bind_bpf *allow_bind_bpf_free(struct allow_bind_bpf *obj) {
+        if (obj && libbpf_get_error(obj) == 0)
+                (void) allow_bind_bpf__destroy(obj);
+
+        return NULL;
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct allow_bind_bpf *, allow_bind_bpf_free);
+
+static int can_link_bpf_program(struct bpf_program *prog) {
+        _cleanup_(bpf_link_freep) struct bpf_link *link = NULL;
+        int err;
+
+        assert(prog);
+
+        link = bpf_program__attach_cgroup(prog, -1);
+        if (!link)
+                return -ENOMEM;
+
+        err = libbpf_get_error(link);
+        if (err != -EINVAL && err != -EBADF)
+                return err;
+
+        /* EBADF indicates that bpf_link is supported by kernel. */
+        return err == -EBADF;
+}
+
+static int prepare_allow_bind_bpf(Unit *u, const Set *ports_v6, const Set *ports_v4, struct allow_bind_bpf **ret_obj) {
+        _cleanup_(allow_bind_bpf_freep) struct allow_bind_bpf *obj = 0;
+        int map_fd, r = 0;
+        uint8_t dummy;
+        void *port;
+
+        obj = allow_bind_bpf__open();
+        if (!obj)
+                return log_unit_error_errno(u, r, "Failed to open BPF object");
```

Comment:
r is not initialized here, i.e second param is effectively zero here

maybe pass `SYNTHETIC_ERRNO(ENOMEM)` here?